gammapy/data/spectral_cube.py

```diff
@@ -413,29 +413,43 @@ def to_fits(self):
         Returns
         -------
         hdu_list : `~astropy.io.fits.HDUList`
-            * hdu_list[0] : `~astropy.io.fits.ImageHDU`
+            * hdu_list[0] : `~astropy.io.fits.PrimaryHDU`
                 Image array of data
             * hdu_list[1] : `~astropy.io.fits.BinTableHDU`
                 Table of energies
         """
-        image = fits.ImageHDU(self.data, self.wcs.to_header())
-        energies = fits.BinTableHDU(self.energy, 'ENERGY')
+        image = fits.PrimaryHDU(self.data.value, self.wcs.to_header())
+        image.header['SPEC_UNIT'] = '{0.unit:FITS}'.format(self.data)
+
+        # for BinTableHDU's the data must be added via a Table object
+        # TODO: this could be simplified a bit by using
```

Comment:
Fixed. I moved the method to `~gammapy.utils.fits.table_to_fits_bin_table` and made it public.